clients/vault/src/metrics.rs

```diff
@@ -432,15 +438,61 @@ async fn publish_average_bitcoin_fee(vault: &VaultData) {
 	vault.metrics.average_btc_fee.gauge.set(average);
 }
 
-// fn publish_bitcoin_balance(vault: &VaultData) {
-//     match vault.btc_rpc.get_balance(None) {
-//         Ok(bitcoin_balance) => vault.metrics.btc_balance.actual.set(bitcoin_balance.to_btc() as
-// f64),         Err(e) => {
-//             // unexpected error, but not critical so just continue
-//             tracing::warn!("Failed to get Bitcoin balance: {}", e);
-//         }
-//     }
-// }
+async fn publish_stellar_balance<P: OraclePallet>(parachain_rpc: P, vault: &VaultData) {
+	match vault.stellar_wallet.read().await.get_balance().await {
+		Ok(balance) => {
+			let currency_id = vault.vault_id.wrapped_currency();
+			let asset: Result<stellar::Asset, _> = currency_id.try_into();
+			let mut wrapped_to_collateral: u128 = 0;
+			if let Ok(asset) = asset {
+				let xlm_balance: Option<f64> = match asset {
+					stellar::Asset::AssetTypeNative => balance
+						.iter()
+						.find(|i| i.asset_type == STELLAR_NATIVE_ASSET_TYPE.to_vec())
+						.map(|i| i.balance),
+					stellar::Asset::AssetTypeCreditAlphanum4(a4) => balance
+						.iter()
+						.find(|i| {
+							i.asset_type != STELLAR_NATIVE_ASSET_TYPE.to_vec() &&
+								i.asset_code.clone().unwrap_or_default() ==
+									a4.asset_code.to_vec()
+						})
+						.map(|i| i.balance),
+					stellar::Asset::AssetTypeCreditAlphanum12(a12) => balance
+						.iter()
+						.find(|i| {
+							i.asset_type != STELLAR_NATIVE_ASSET_TYPE.to_vec() &&
+								i.asset_code.clone().unwrap_or_default() ==
+									a12.asset_code.to_vec()
+						})
+						.map(|i| i.balance),
+					_ => {
+						tracing::warn!("Unsupported stellar asset type");
+						None
+					},
```

Comment:
Maybe we can extract this whole logic of extracting the balance of a specific asset from a vec of balances to a separate function? Should make the code easier to read

Reply:
Done. Resolved!